edx_event_bus_kafka/internal/consumer.py

```diff
@@ -129,10 +130,38 @@ def _shut_down(self):
         """
         self._shut_down_loop = True
 
-    def consume_indefinitely(self):
+    def consume_indefinitely(self, offset_timestamp=None):
         """
         Consume events from a topic in an infinite loop.
+
+        Arguments:
+            offset_timestamp(datetime): reset the offsets of the consumer partitions to this timestamp before consuming.
         """
+
+        def reset_offsets(consumer, partitions):
+            # This is a callback method used on consumer assignment to handle offset reset logic.
+            # We do not want to attempt to change offsets if the offset is None.
+            if offset_timestamp is None:
+                return
+
+            # Get the offset from the epoch in seconds.
+            offset_timestamp_s = int(offset_timestamp.timestamp())
+            # We construct partitions with the epoch timestamp in the offset position.
+            timestamp_partitions = [TopicPartition(self.topic, p.partition, offset_timestamp_s) for p in partitions]
+
+            partitions_with_offsets = consumer.offsets_for_times(timestamp_partitions, timeout=1.0)
+
+            # Partitions have an error field that may be set on return.
+            errors = [p.error for p in partitions_with_offsets if p.error is not None]
+            if len(errors) > 0:
+                for error in errors:
+                    logger.error(f"Error getting offsets for timestamps: {error}")
+                raise Exception("Error getting offsets for timestamps")
```

Comment:
Maybe just collapse it down to this?
```suggestion
                raise Exception("Error getting offsets for timestamps: {errors}")
```